src/io/test.zig

```diff
@@ -23,31 +23,67 @@ test "write/read/close" {
 
         fn run_test() !void {
             const path = "test_io_write_read_close";
-            const file = try std.fs.cwd().createFile(path, .{ .read = true, .truncate = true });
+
+            // The file gets created below, either by createFile or openat.
             defer std.fs.cwd().deleteFile(path) catch {};
 
             var self: Context = .{
                 .io = try IO.init(32, 0),
-                .fd = file.handle,
             };
             defer self.io.deinit();
 
             var completion: IO.Completion = undefined;
 
+            if (builtin.target.os.tag == .linux) {
+                self.io.openat(
+                    *Context,
+                    &self,
+                    openat_callback,
+                    &completion,
+                    os.AT.FDCWD,
+                    path,
+                    os.O.RDWR | os.O.TRUNC | os.O.CREAT,
+                    std.fs.File.default_mode,
+                );
+            } else {
+                const file = try std.fs.cwd().createFile(path, .{
+                    .read = true,
+                    .truncate = true,
+                });
+                self.fd = file.handle;
+
+                self.io.write(
+                    *Context,
+                    &self,
+                    write_callback,
+                    &completion,
+                    self.fd.?,
+                    &self.write_buf,
+                    10,
+                );
```

Comment:
Would it be nicer to do
```suggestion
                self.openat_callback(&completion, file.handle)
```

Reply:
Much!